openfl-docker/Dockerfile.base

```diff
@@ -1,122 +1,33 @@
-# Copyright (C) 2020-2023 Intel Corporation
+# Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
+# ------------------------------------
+# OpenFL Base Image
+# ------------------------------------
+FROM ubuntu:22.04 as base
 
-# If your machine is behind a proxy, make sure you set it up in ~/.docker/config.json
-
-ARG IMAGE_NAME=ubuntu
-ARG IMAGE_TAG=22.04
-
-# Base image to be used everywhere
-FROM ${IMAGE_NAME}:${IMAGE_TAG} as base
-RUN apt-get clean && \
-    apt-get update && \
-    apt-get upgrade -y && \
-    apt-get clean autoclean && \
-    apt-get autoremove -y && \
-    rm -rf /var/lib/apt/lists/*
-
-# Python image
-FROM base as python-dev
-
-ARG PYTHON=python3.10
-ARG PIP=pip3.10
-
-RUN apt-get update && \
-    apt-get install --no-install-recommends --fix-missing -y \
-      curl \
-      gpg-agent \
-      software-properties-common && \
-    add-apt-repository -y ppa:deadsnakes/ppa && \
-    apt-get update && \
-    DEBIAN_FRONTEND=noninteractive \
-    apt-get install --no-install-recommends --fix-missing -y \
-      ${PYTHON} \
-      ${PYTHON}-distutils && \
-    curl -s https://bootstrap.pypa.io/get-pip.py | ${PYTHON} && \
-    apt-get purge -y \
-      curl \
-      gpg-agent \
-      software-properties-common && \
-    apt-get autoclean && \
-    apt-get autoremove -y && \
-    rm -rf /var/lib/apt/lists/*
-
-RUN ln -sf /usr/bin/${PYTHON} /usr/bin/python && \
-    ln -sf /usr/bin/${PYTHON} /usr/bin/python3 && \
-    ln -sf /usr/local/bin/${PIP} /usr/local/bin/pip && \
-    ln -sf /usr/local/bin/${PIP} /usr/local/bin/pip3
-
-FROM base as python-base
-
-ARG PYTHON=python3.10
-
-COPY --from=python-dev /usr/lib/x86_64-linux-gnu /usr/lib/x86_64-linux-gnu
-COPY --from=python-dev /usr/lib/${PYTHON} /usr/lib/${PYTHON}
-COPY --from=python-dev /usr/local/lib/${PYTHON} /usr/local/lib/${PYTHON}
-COPY --from=python-dev /usr/bin /usr/bin
-COPY --from=python-dev /usr/local/bin /usr/local/bin
-
-FROM python-base as openfl
-
+USER root
```

Comment:
I think this will continue to trigger the Coverity security issue: _The Docker container is configured to run as the root user._ Is it possible to set this to run as a non-root user? Then we can mirror that in the gramine Dockerfile

Reply:
Done.